File: algorithms/sorting/selection-sort/c/selection_sort.c
#include "selection_sort.h"

static void swap(int* a, int* b) {
    int t = *a;
    *a = *b;
    *b = t;
}

void selection_sort(int arr[], int n) {
    int i, j, min_idx;
    for (i = 0; i < n - 1; i++) {
        min_idx = i;
        for (j = i + 1; j < n; j++) {
            if (arr[j] < arr[min_idx])
                min_idx = j;
        }
        swap(&arr[min_idx], &arr[i]);
    }
}

File: algorithms/sorting/selection-sort/rust/selection_sort.rs
pub fn selection_sort(arr: &mut [i32]) {
    let n = arr.len();
    if n == 0 {
        return;
    }
    
    for i in 0..n-1 {
        let mut min_idx = i;
        for j in i+1..n {
            if arr[j] < arr[min_idx] {
                min_idx = j;
            }
        }
        arr.swap(min_idx, i);
    }
}
